airbyte-integrations/connectors/source-redshift/src/main/java/io/airbyte/integrations/source/redshift/RedshiftSource.java

```diff
@@ -93,6 +97,31 @@ private void addSsl(final List<String> additionalProperties) {
     return Set.of("information_schema", "pg_catalog", "pg_internal", "catalog_history");
   }
 
+  @Override
+  public Set<JdbcPrivilegeDto> getPrivilegesTableForCurrentUser(final JdbcDatabase database, final String schema) throws SQLException {
+    // TODO (liren): The bufferedResultSetQuery method is used here instead of the
+    // query method. This is because the latter close the connection in the onClose method
+    // of the stream it returns. However, sometimes the onClose method is somehow never
+    // executed. This seems to happen when the result is empty from Redshift. Issue #10128.
```

Comment:
This comment can be removed. This hanging issue was caused by unclosed connection. The root cause has been resolved.

> This seems to happen when the result is empty from Redshift.
My previous diagnosis was not true.

```suggestion
```